### PR TITLE
Update release and publish workflows

### LINKED ISSUE #85: Release and publish workflows should be separated from each other
Currently, the [release.yaml](https://github.com/bakdata/kpops/blob/main/.github/workflows/release.yaml) file publishes the Python packages to PyPi **and then** creates a git tag, GitHub release, and updates/pushes the docs.
We should change the order and separate the release.yaml into to workflows: release, publish.

## Release Wrokflow
This workflow should be triggered manually, and it is similar to the current [release.yaml](https://github.com/bakdata/kpops/blob/main/.github/workflows/release.yaml) workflow. Except it only creates a git tag and creates a GitHub release

## Publish Workflow
This workflow will be **triggered only if a new tag is pushed**. This workflow will publish the Python packages to PyPi and update/pubish the docs to GitHub pages.

### PR DESCRIPTION
**Pull request by irux**
Closes #85

### CODE REVIEW

**Comment by irux** [comment]
@raminqaf @disrupted how would you label this PR in order to appear in the changelog ? should I add all `area` labels to the changelog config ?

**Comment by disrupted** [comment]
> @raminqaf @disrupted how would you label this PR in order to appear in the changelog ? should I add all `area` labels to the changelog config ?

looks good to me as it is

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -14,43 +14,18 @@ on:
...
-  release-to-pypi:
-    uses: bakdata/ci-templates/.github/workflows/python-poetry-release.yaml@v1.18.2
+  create-github-release-push-tag:
+    runs-on: ubuntu-22.04
+    uses: bakdata/ci-templates/.github/workflows/python-poetry-release.yaml@v1.22.0
     with:
-      release-type: ${{ github.event.inputs.release-type }}
-      publish-to-test: false
-      python-version: "3.10"
+      release-type: ${{inputs.release-type}}
```
Comment (disrupted) [inline .github/workflows/release.yaml:21]: surrounding space for consistency
```suggestion
      release-type: ${{ inputs.release-type }}
```

**Comment by disrupted** [comment]
tags will from now on be created without the leading `v`, right?

**Comment by irux** [comment]
> tags will from now on be created without the leading `v`, right?

yes! that's how we build the action :)

**Comment by irux** [comment]
> > @raminqaf @disrupted how would you label this PR in order to appear in the changelog ? should I add all `area` labels to the changelog config ?
> 
> looks good to me as it is

so, how should we label this PR ?

**Comment by disrupted** [comment]
> 

`area/CI` is good, it doesn't even have to be part of the changelog. it's not relevant to kpops users

**Comment by sujuka99** [comment]
closes #85 ?

**Review by sujuka99 — APPROVED** [review]
LGTM

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -14,43 +14,18 @@ on:
           - major
 
 jobs:
-  release-to-pypi:
-    uses: bakdata/ci-templates/.github/workflows/python-poetry-release.yaml@v1.18.2
+  create-github-release-push-tag:
+    runs-on: ubuntu-22.04
+    uses: bakdata/ci-templates/.github/workflows/python-poetry-release.yaml@v1.22.0
```
Comment (raminqaf) [inline .github/workflows/release.yaml:19]:
```suggestion
    name: Publish to PyPI
    uses: bakdata/ci-templates/.github/workflows/python-poetry-release.yaml@v1.22.0
```

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -64,9 +64,9 @@ jobs:
       - name: Test docs build (mkdocs)
         run: poetry run mkdocs build -f docs/mkdocs.yml
 
-  publish:
-    name: Publish
+  publish-snapshot-version:
+    name: Publish Snapshot to TestPyPI
```
Comment (disrupted) [inline .github/workflows/ci.yaml:68]:
```suggestion
    name: Publish snapshot to TestPyPI
```

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -14,43 +14,19 @@ on:
           - major
 
 jobs:
-  release-to-pypi:
-    uses: bakdata/ci-templates/.github/workflows/python-poetry-release.yaml@v1.18.2
+  create-github-release-push-tag:
+    runs-on: ubuntu-22.04
+    uses: bakdata/ci-templates/.github/workflows/python-poetry-release.yaml@v1.23.0
+    name: build project and create GitHub release
```
Comment (disrupted) [inline .github/workflows/release.yaml:20]: this doesn't "build the project"
```suggestion
    name: Release
```

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -14,43 +14,19 @@ on:
...
-        with:
-          tag_name: "${{ needs.release-to-pypi.outputs.release-tag }}"
-          name: Release v${{ needs.release-to-pypi.outputs.release-tag }}
-          draft: false
-          prerelease: false
+      github-username: "${{ secrets.GH_USERNAME }}"
+      github-email: "${{ secrets.GH_EMAIL }}"
+      github-token: "${{ secrets.GH_TOKEN }}"
 
   update-docs:
```
Comment (disrupted) [inline .github/workflows/release.yaml:31]: `needs` must be updated

File: .github/workflows/release.yaml
Comment (disrupted) [inline .github/workflows/release.yaml:1]: update workflow name

**Review by disrupted — CHANGES_REQUESTED** [review]

**Review by disrupted — CHANGES_REQUESTED** [review]

**Review by disrupted — APPROVED** [review]

**Pull request merged**